src/utils/daemon.rs

```diff
@@ -18,17 +18,15 @@ use time;
 use ::{libc::fork, std::fs::File, std::io::Write, std::process::exit};
 
 pub fn start_daemon(background: bool) {
-    // first check required environment variables
-    for v in [
+    const CRATE_VARIABLES: [&str; 3] = [
         "CRATESFYI_PREFIX",
         "CRATESFYI_GITHUB_USERNAME",
         "CRATESFYI_GITHUB_ACCESSTOKEN",
-    ]
-    .iter()
-    {
-        if env::var(v).is_err() {
-            panic!("Environment variable {} not found", v);
-        }
```

Comment:
I prefer the way it was before, I think it was easier to read. Also I don't see the point of having a `const` since it's only used once, and that directly next to where the array is defined.

Reply:
I think it's much cleaner this way because of how it removes the rather large array away from the for statement

Reply:
The const also just gives some more guarantees that it'll be flattened in the final code

Reply:
> I think it's much cleaner this way because of how it removes the rather large array away from the for statement

Ok, can you change it to just being a `const` array instead of `unwrap_or_else`? The first time I read the unwrap_or_else I was worried it panicked eagerly instead of on error :sweat_smile: 

> The const also just gives some more guarantees that it'll be flattened in the final code

Can you expand on this? My understanding is that for any `const X = blah;`, `X` behaves exactly as if you'd typed `blah` in the source code (modulo bugs in rustc).

Reply:
> Ok, can you change it to just being a `const` array instead of `unwrap_or_else`? The first time I read the unwrap_or_else I was worried it panicked eagerly instead of on error 😅
Sure thing, it's actually better as the if now that I look at it

> Can you expand on this? My understanding is that for any` const X = blah;`, `X` behaves exactly as if you'd typed` blah` in the source code (modulo bugs in rustc).
Yah, that's exactly what I mean, the `const` makes sure that the variable is "the same" which could possibly not happen with a `let`. Adding onto the `const` train, if ever more variables are added then we don't end up with the longest for statement in the world, we just extend the const some. It gives more clarity to the code imo, but that's a personal thing